src/mgr/OSDPerfMetricCollector.h

```diff
@@ -0,0 +1,60 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+#ifndef OSD_PERF_METRIC_COLLECTOR_H_
+#define OSD_PERF_METRIC_COLLECTOR_H_
+
+#include "include/utime.h"
+
+#include "common/Mutex.h"
+
+#include <list>
+#include <map>
+
+// XXXMG: this is going to be somewhere else
+struct OSDPerfMetricQuery {
+  bool operator<(const OSDPerfMetricQuery &other) const {
+    return false;
+  }
+};
+inline std::ostream& operator<<(std::ostream& os, const OSDPerfMetricQuery &query) {
+  return os << "simple";
+}
+// XXXMG end
+
+/**
+ * OSD performance query class.
+ */
+class OSDPerfMetricCollector {
+public:
+  typedef std::map<OSDPerfMetricQuery, std::map<uint64_t, utime_t>>
```

Comment:
It would be good to have a typedef'd name for the ID type, rather than using uint64_t directly everywhere, for readability and in case we ever change it.